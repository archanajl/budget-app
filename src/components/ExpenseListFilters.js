import 'react-dates/initialize';
import React from 'react';
import {connect} from 'react-redux';
import {setTextFilter, sortByDate,sortByAmount,setStartDate,setEndDate} from '../actions/filters';
import {DateRangePicker} from 'react-dates';

export class ExpenseListFilters extends React.Component{

    state = {
        calendarFocused : null
    };

    onDatesChange =({startDate,endDate}) =>{
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    }

    onFocusChange = (focused) => {
        this.setState(()=>({calendarFocused : focused}))
    }

    onTextChange=(e) =>{
        this.props.setTextFilter(e.target.value);
    }

    onSelectChange= (e)=>{
        if (e.target.value==="date"){
            this.props.sortByDate();
        }
        else{
            this.props.sortByAmount();
        }
    }

    render(){
        return (
            <div>
                <div className="container">
                    <h4>Filter Text :</h4>
                    <input type ="text" value={this.props.filters.text} onChange={this.onTextChange}></input>
                    <h4>Sort By : </h4>
                    <select value={this.props.filters.sortBy} onChange={this.onSelectChange}>
                        <option value="date">Date</option>
                        <option value="amount">Amount</option>
                    </select>
                <DateRangePicker
                    startDate={this.props.filters.startDate}
                    endDate = {this.props.filters.endDate}
                    onDatesChange = {this.onDatesChange}
                    focusedInput ={this.state.calendarFocused}
                    onFocusChange = {this.onFocusChange}
                    showClearDates = {true}
                    numberOfMonths ={1}
                    isOutsideRange ={()=>false}
                />
                </div>
            </div>
        )
    }
} 

const mapStateToProps = (state) => {
    return {
        filters : state.filters
    }
}

const mapDispatchToProps = (dispatch) => ({
    setStartDate : (startDate) => dispatch(setStartDate(startDate)),
    setEndDate : (endDate) => dispatch(setEndDate(endDate)),
    setTextFilter : (textValue) => dispatch(setTextFilter(textValue)),
    sortByDate : () => dispatch(sortByDate()),
    sortByAmount : () => dispatch(sortByAmount())

})

export default connect(mapStateToProps,mapDispatchToProps)(ExpenseListFilters);