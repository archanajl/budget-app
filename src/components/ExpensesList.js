import React from 'react';
import {connect} from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses'

export const ExpensesList = (props) =>(
    <div>
        <h1>
            {props.expenses.map((expense) => <ExpenseListItem key = {expense.id} {...expense}/>)}
        </h1>
    </div>
)

const mapToStateProps = (state) => {
    return {
        expenses : selectExpenses(state.expenses,state.filters)
    }
}

export default connect(mapToStateProps)(ExpensesList);