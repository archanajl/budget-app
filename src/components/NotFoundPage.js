import React from 'react'

const NotFoundPage = () => (
    <div> 404 !! Not found Page </div>
)

export default NotFoundPage;