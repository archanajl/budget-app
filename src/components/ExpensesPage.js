import React from 'react';
import ExpensesList from './ExpensesList';
import ExpenseListFilters from './ExpenseListFilters';


const ExpensesPage = () => (
    <div>
        <ExpenseListFilters/> 
        <ExpensesList/>
    </div>
)

export default ExpensesPage;