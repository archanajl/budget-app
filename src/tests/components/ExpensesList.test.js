import React from 'react';
import {render, shallow} from 'enzyme';
import {ExpensesList} from '../../components/ExpensesList';
import expenses from '../fixtures/expenses';

test('should render the expenses List with expenses',()=>{
    const wrapper = shallow(<ExpensesList expenses={expenses} />)
    expect(wrapper).toMatchSnapshot();
})

test('should render the expenses List with empty expenses',()=>{
    const wrapper = shallow(<ExpensesList expenses={[]} />)
    expect(wrapper).toMatchSnapshot();
})