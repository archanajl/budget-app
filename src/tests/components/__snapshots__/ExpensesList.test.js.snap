// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the expenses List with empty expenses 1`] = `
<div>
  <h1 />
</div>
`;

exports[`should render the expenses List with expenses 1`] = `
<div>
  <h1>
    <ExpenseListItem
      amount={100}
      createdAt={0}
      description="water bill"
      id="1"
      key="1"
      note="bill"
    />
    <ExpenseListItem
      amount={200}
      createdAt={-345600000}
      description="gas bill"
      id="2"
      key="2"
      note="bill"
    />
    <ExpenseListItem
      amount={300}
      createdAt={345600000}
      description="rent bill"
      id="3"
      key="3"
      note="bill"
    />
  </h1>
</div>
`;
